src/toil/test/__init__.py

```diff
@@ -140,20 +140,41 @@ def needs_aws(test_item):
         dot_boto_path = os.path.expanduser('~/.boto')
         dot_aws_credentials_path = os.path.expanduser('~/.aws/credentials')
         hv_uuid_path = '/sys/hypervisor/uuid'
-        if (os.path.exists(dot_boto_path)
-            or os.path.exists(dot_aws_credentials_path)
-            # Assume that EC2 machines like the Jenkins slave that we run CI on will have IAM roles
-            or os.path.exists(hv_uuid_path) and file_begins_with(hv_uuid_path,'ec2')):
+        try:
+            with open(dot_boto_path) as f:
+                if 'aws_access_key_id' in f.read().decode('utf8'):
+                    return test_item
+        except IOError:
```

Comment:
Should be more specific. Only errno.ENOENT is ok everything else should be reraised.